File: apps/server/src/cocktail_server/schemas/images.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict

from cocktail_server.schemas.generate import AspectRatio, CfgPreset


class ImageUploadResponse(BaseModel):
    """`POST /images` の応答。返ってきた `image_id` を ImagePart に詰めて chat に送る。"""

    model_config = ConfigDict(strict=True, extra="forbid")

    image_id: str
    image_url: str
    mime: str
    width: int
    height: int


class GeneratedImageRef(BaseModel):
    """`generate_image` ツールで生成された画像のメタデータ。ギャラリー横断ビュー用。"""

    model_config = ConfigDict(strict=True, extra="forbid")

    image_id: str
    image_url: str
    conversation_id: str
    created_at: datetime
    prompt: str
    seed: int
    aspect_ratio: AspectRatio
    cfg_preset: CfgPreset
    width: int
    height: int


class GeneratedImageList(BaseModel):
    """`GET /images` の応答。`next_before` を `?before=` に渡すと次ページが取れる。"""

    model_config = ConfigDict(strict=True, extra="forbid")

    images: list[GeneratedImageRef]
    next_before: datetime | None = None

File: packages/api-types/src/images.ts
import { z } from "zod";

/** `POST /images` のレスポンス。返ってきた `image_id` を ImagePart に詰めて chat に送る。 */
export const ImageUploadResponseSchema = z
  .object({
    image_id: z.string(),
    image_url: z.string(),
    mime: z.string(),
    width: z.number().int(),
    height: z.number().int(),
  })
  .strict();

export type ImageUploadResponse = z.infer<typeof ImageUploadResponseSchema>;

/** ギャラリー一覧で返る 1 件分の生成画像メタ。`created_at` は ISO8601 文字列。 */
export const GeneratedImageRefSchema = z
  .object({
    image_id: z.string(),
    image_url: z.string(),
    conversation_id: z.string(),
    created_at: z.string(),
    prompt_excerpt: z.string(),
    seed: z.number().int(),
    aspect_ratio: z.enum(["portrait", "landscape", "square"]),
    cfg_preset: z.enum(["soft", "standard", "crisp"]),
    width: z.number().int(),
    height: z.number().int(),
  })
  .strict();

export type GeneratedImageRef = z.infer<typeof GeneratedImageRefSchema>;

/** `GET /images` のレスポンス。`next_before` を `?before=` に渡すと続きが取れる。 */
export const GeneratedImageListSchema = z
  .object({
    images: z.array(GeneratedImageRefSchema),
    next_before: z.string().nullable(),
  })
  .strict();

export type GeneratedImageList = z.infer<typeof GeneratedImageListSchema>;
